src/nomad_measurements/xrd/nx.py

```diff
@@ -0,0 +1,264 @@
+            template['/ENTRY[entry]/experiment_result/omega/@units'] = archive.data.results[0].omega.units.__str__()
+        except AttributeError as e:
+            pass
+
+        try:
+            template['/ENTRY[entry]/experiment_result/chi'] = archive.data.results[0].chi.magnitude
+            template['/ENTRY[entry]/experiment_result/chi/@units'] = archive.data.results[0].chi.units.__str__()
+        except AttributeError as e:
+            pass
+
+        try:
+            template['/ENTRY[entry]/experiment_result/phi'] = archive.data.results[0].phi.magnitude
+            template['/ENTRY[entry]/experiment_result/phi/@units'] = archive.data.results[0].phi.units.__str__()
+        except AttributeError as e:
+            pass
+
+        try:
+            template[
+                '/ENTRY[entry]/INSTRUMENT[instrument]/DETECTOR[detector]/scan_axis'
+            ] = archive.data.results[0].scan_axis
+        except AttributeError as e:
+            pass
+
+        try:
+            template['/ENTRY[entry]/experiment_config/count_time'] = archive.data.results[0].count_time.magnitude
+        except AttributeError as e:
+            pass
+    # rsm
+    elif scan_type == 'rsm':
+        try:
+            template['/ENTRY[entry]/experiment_result/intensity'] = archive.data.results[0].intensity.magnitude
+        except AttributeError as e:
+            pass
+
+        try:
+            template['/ENTRY[entry]/experiment_result/two_theta'] = archive.data.results[0].two_theta.magnitude
+            template['/ENTRY[entry]/experiment_result/two_theta/@units'] = archive.data.results[0].two_theta.units.__str__()
+        except AttributeError as e:
+            pass
+
+        try:
+            template['/ENTRY[entry]/experiment_result/omega'] = archive.data.results[0].omega.magnitude
+            template['/ENTRY[entry]/experiment_result/omega/@units'] = archive.data.results[0].omega.units.__str__()
+        except AttributeError as e:
+            pass
+
+        try:
+            template['/ENTRY[entry]/experiment_result/chi'] = archive.data.results[0].chi.magnitude
+            template['/ENTRY[entry]/experiment_result/chi/@units'] = archive.data.results[0].chi.units.__str__()
+        except AttributeError as e:
+            pass
+
+        try:
+            template['/ENTRY[entry]/experiment_result/phi'] = archive.data.results[0].phi.magnitude
+            template['/ENTRY[entry]/experiment_result/phi/@units'] = archive.data.results[0].phi.units.__str__()
+        except AttributeError as e:
+            pass
+
+        try:
+            template[
+                '/ENTRY[entry]/INSTRUMENT[instrument]/DETECTOR[detector]/scan_axis'
+            ] = archive.data.results[0].scan_axis
+        except AttributeError as e:
+            pass
+
+        try:
+            template['/ENTRY[entry]/experiment_config/count_time'] = archive.data.results[0].count_time.magnitude
+        except AttributeError as e:
+            pass
```

Comment:
```suggestion
```

These quantities are common for `rsm` and `line` scans. You can take it out of the if clause,